rootfs/etc/haproxy/template/haproxy.tmpl

```diff
@@ -87,8 +87,14 @@ frontend httpfront
 {{ if ne $server.Hostname "_" }}
 {{ if isWildcardHostname $server.Hostname }}
     acl ishost-{{ labelize $server.Hostname }} hdr_reg(host) {{ hostnameRegex $server.Hostname }} {{ hostnameRegex $server.Hostname }}:80
+{{ if ne $server.Alias "" }}
+    acl ishost-{{ labelize $server.Hostname }}-alias {{ if isRegexpHostname $server.Alias }}hdr_reg(host) {{ aliasRegex $server.Alias }}{{ else }}hdr(host) {{ $server.Alias }}{{ end }} {{ hostnameRegex $server.Hostname }}:80
+{{ end }}
 {{ else }}
     acl ishost-{{ labelize $server.Hostname }} hdr(host) {{ $server.Hostname }} {{ $server.Hostname }}:80
+{{ if ne $server.Alias "" }}
+    acl ishost-{{ labelize $server.Hostname }}-alias {{ if isRegexpHostname $server.Alias }}hdr_reg(host) {{ aliasRegex $server.Alias }}{{ else }}hdr(host) {{ $server.Alias }}{{ end }} {{ $server.Hostname }}:80
+{{ end }}
```

Comment:
These two `if ne $serverAlias ""` could be outside `if isWildcard` condition. The `:80` should also use the `$serverAlias`.

Note also that you shoud quote your regex or escape spaces, otherwise HAProxy will read two broken regex.

Reply:
I haven't decided yet what I should do with `:80` at the end of ACL. If alias is used as a regex then `:80` is supposed to be a part of regex. But what if I have a dollar sign is placed at the end of the regex?

Reply:
oh, I've found what you made in 917bb1032df09d0b7607e7f040161a595510ac8b. Ok then.